src/languages/cpp.js

```diff
@@ -67,27 +67,27 @@ function(hljs) {
   var FUNCTION_TITLE = hljs.IDENT_RE + '\\s*\\(';
 
   var CPP_KEYWORDS = {
-    keyword: 'int float while private char catch import module export virtual operator sizeof ' +
+    keyword: 'int float while private char char8_t char16_t char32_t catch import module export virtual operator sizeof ' +
       'dynamic_cast|10 typedef const_cast|10 const for static_cast|10 union namespace ' +
       'unsigned long volatile static protected bool template mutable if public friend ' +
-      'do goto auto void enum else break extern using asm case typeid ' +
+      'do goto auto void enum else break extern using asm case typeid wchar_t' +
       'short reinterpret_cast|10 default double register explicit signed typename try this ' +
-      'switch continue inline delete alignof constexpr consteval constinit decltype ' +
+      'switch continue inline delete alignas alignof constexpr consteval constinit decltype ' +
       'concept co_await co_return co_yield requires ' +
-      'noexcept static_assert thread_local restrict _Bool complex _Complex _Imaginary ' +
+      'noexcept static_assert thread_local restrict final override ' +
       'atomic_bool atomic_char atomic_schar ' +
       'atomic_uchar atomic_short atomic_ushort atomic_int atomic_uint atomic_long atomic_ulong atomic_llong ' +
       'atomic_ullong new throw return ' +
-      'and or not',
+      'and and_eq bitand bitor compl not not_eq or or_eq xor xor_eq',
     built_in: 'std string wstring cin cout cerr clog stdin stdout stderr stringstream istringstream ostringstream ' +
       'auto_ptr deque list queue stack vector map set bitset multiset multimap unordered_set ' +
-      'unordered_map unordered_multiset unordered_multimap array shared_ptr abort abs acos ' +
+      'unordered_map unordered_multiset unordered_multimap array shared_ptr abort terminate abs acos ' +
       'asin atan2 atan calloc ceil cosh cos exit exp fabs floor fmod fprintf fputs free frexp ' +
       'fscanf future isalnum isalpha iscntrl isdigit isgraph islower isprint ispunct isspace isupper ' +
       'isxdigit tolower toupper labs ldexp log10 log malloc realloc memchr memcmp memcpy memset modf pow ' +
       'printf putchar puts scanf sinh sin snprintf sprintf sqrt sscanf strcat strchr strcmp ' +
       'strcpy strcspn strlen strncat strncmp strncpy strpbrk strrchr strspn strstr tanh tan ' +
-      'vfprintf vprintf vsprintf endl initializer_list unique_ptr',
+      'vfprintf vprintf vsprintf endl initializer_list unique_ptr _Bool complex _Complex _Imaginary',
```

Comment:
Shouldn't we also have `imaginary` as well?

Reply:
Surprisingly, no. Those are keywords coming from C99, not all of them have their equivalents in the C++ standard library.

Reply:
I said that because of: https://en.cppreference.com/w/c/numeric/complex/imaginary

```
    double imaginary i = 2.0*I; // pure imaginary
```

> This macro expands to the keyword _Imaginary.

Defined in complex.h... if we don't count both how are we deciding that _Imaginary is a keyword but imaginary is not?

I'm welcome to be wrong here, but I'm trying to understand the logic we're using to call something a keyword.

Reply:
You didn't respond?  Am I thinking about this wrong? ... ?

Reply:
```
double imaginary i = 2.0*I; // pure imaginary
double _Imaginary i = 2.0*I; // pure imaginary
```

Given this snippet, wouldn't both cases of imaginary there count as keywords? Yes/no?  Or are we drawing a distinction between build-in vs pulled in via some include?

Complex seems to be in the same boat:

https://en.cppreference.com/w/c/language/arithmetic_types#Complex_floating_types

So it seems to me if `complex` is a keyword, then `imaginary` also should be, and if not, then it should not?

Reply:
I think if you just want to do the simplest thing to get this PR approved from where I'm setting it seems we need to add `imaginary` as a keyword also, and then we're consistent. But fell free to make a case the other way if you think that's not correct.